client-sdk/go/types/token.go

```diff
@@ -87,18 +87,38 @@ func NewBaseUnits(amount quantity.Quantity, denomination Denomination) BaseUnits
 	}
 }
 
+// FormatNamedAddress returns a human-friendly representation of an address:
+// it prints the name (if known) followed by the preferred form of the address
+// in parentheses. If an Ethereum hex address mapping is provided for the native
+// address, it is used; otherwise the native Bech32 address is used.
+func FormatNamedAddress(ctx context.Context, addr Address) string {
```

Comment:
```suggestion

// FormatNamedAddress returns a human-friendly representation of an address.
//
// It prints the name (if known) followed by the preferred form of the address
// in parentheses. If an Ethereum hex address mapping is provided for the native
// address, it is used; otherwise the native Bech32 address is used.
func FormatNamedAddress(ctx context.Context, addr Address) string {
	native := addr.String()

	name, ok := getAccountName(ctx, native)
	if !ok {
		return native
	}

	hex, ok := getEthereumHex(ctx, native)
	if !ok {
		return fmt.Sprintf("%s (%s)", name, native)
	}

	return fmt.Sprintf("%s (%s)", name, hex)
}

func getAccountName(ctx context.Context, addr string) (string, bool) {
	names, ok := ctx.Value(ContextKeyAccountNames).(AccountNames)
	if !ok {
		return "", false
	}
	name, ok := names[addr]
	return name, ok
}

func getEthereumHex(ctx context.Context, addr string) (string, bool) {
	addrs, ok := ctx.Value(ContextKeyAccountEthMap).(map[string]string)
	if !ok {
		return "", false
	}
	hex, ok := addrs[addr]
	return hex, ok
}
```
I prefer always using `ok`, not `exists`, `has`, `found`, ...